### PR TITLE
stacks: install should omit the tag from stack names derived from image names

### LINKED ISSUE #22: stack install fails for stacks with an image tag
STACK_NAME appears to be optional in the stack install help.  This is reasonable as a default STACK_NAME could be derived from the image name presented in the STACK_IMAGE_NAME (org/name:version).

```
$ kubectl crossplane stack install --help
Usage: kubectl crossplane stack install [-h|--help] [-c|--cluster] STACK_IMAGE_NAME [STACK_NAME [STACK_IMAGE_SOURCE]]
...
```

However, STACK_NAME appears to be required when using real-world images.  I can see how local image names would not contain slashes ("/") and could potentially omit a version, so STACK_NAME may technically be optional today in some limited cases.

```
$ kubectl crossplane stack install --cluster crossplane/stack-gcp:v0.2.0
The ClusterStackInstall "crossplane-stack-gcp:v0.2.0" is invalid: metadata.name: Invalid value: "crossplane-stack-gcp:v0.2.0": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

### PR DESCRIPTION
**Pull request by displague**
Fixes #22

### CODE REVIEW

File: bin/kubectl-crossplane-stack-install [hunk 1]
```diff
@@ -75,7 +75,8 @@ STACK_IMAGE_NAME="${1}"
 # For kubernetes fields, we aren't able to use slashes, and
 # slashes are common for docker image names. So we remove the
 # slashes before we use the name for kubernetes resource fields.
-KUBEY_STACK_IMAGE_NAME=$( echo "${STACK_IMAGE_NAME}" | tr '/' '-' )
+# We also can't use colons and don't want to include tags in the name.
+KUBEY_STACK_IMAGE_NAME=$( echo "${STACK_IMAGE_NAME}" | tr '/' '-' | sed 's/:.*//')
```
Comment (suskin) [inline bin/kubectl-crossplane-stack-install]: Looks like it doesn't quite work yet. With `set -x` enabled:

```bash
$ ./bin/kubectl-crossplane-stack-install crossplane/stack-gcp:master
+ CLUSTER_STACK=
+ CLUSTER_OPT=
+ NAMESPACE_OPT=
+ POSITIONAL=()
+ [[ 1 -gt 0 ]]
+ opt=crossplane/stack-gcp:master
+ case $opt in
+ POSITIONAL+=("$1")
+ shift
+ [[ 0 -gt 0 ]]
+ '[' 1 -eq 0 ']'
+ set -- crossplane/stack-gcp:master
+ STACK_IMAGE_NAME=crossplane/stack-gcp:master
++ echo crossplane/stack-gcp:master
++ tr / -
++ sed 's/:.*//'
+ KUBEY_STACK_IMAGE_NAME=crossplane-stack-gcp
+ STACK_NAME=crossplane-stack-gcp
+ kubectl crossplane stack generate-install crossplane/stack-gcp:master
+ kubectl apply -f -
The StackInstall "crossplane-stack-gcp:master" is invalid: metadata.name: Invalid value: "crossplane-stack-gcp:master": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Looks like we would need to put the same logic into `generate_install`. Or, even better, we could delete `generate_install` now that `install` supports namespace : )

**Review by suskin — CHANGES_REQUESTED** [review]
Requesting changes since it doesn't function correctly yet

File: bin/kubectl-crossplane-stack-install (hunk 1 above)
Comment (displague) [reply bin/kubectl-crossplane-stack-install]: `generate_install` is still useful for running with `| kubectl --context=... apply -f -`

For whatever reason, `kubectl --context=... crossplane` isn't recognized.

Comment (displague) [reply bin/kubectl-crossplane-stack-install]: I've [pushed an update](https://github.com/crossplane/crossplane-cli/compare/743dde4bc7ed0b18eeabc629638748ccdf67def2..2783bae2146f4f51171543a7fc649729a08b3a41) for `generate-install` and added a space before a closing parenthesis.

With these changes, I wasn't able to reproduce the error you received with the command you used. I tried on Linux and OS/X.

**Review by suskin — APPROVED** [review]
Looks good to me. Thanks @displague ! :squid:

**Pull request merged**